### PR TITLE
Add logic to create a jenkins build item

### PR DESCRIPTION
**Pull request by DennisDenuto**
Added logic to create a new jenkins job item.

### CODE REVIEW

**Comment by yosida95** [comment]
Hi,

In Jenkins ver 1.624 (the current latest version), this patch seems not worked.
You can make sure some-test-job existence after running tests?

In addition, test code for CreateJob are weak because
- returned value from CreateJob (response string, err error) are ignored
- counting jobs got by GetJobs will not always ensure job created.
    - In test code context, you have to confirm "some-test-job" is contained in GetJobs response.

**Comment by DennisDenuto** [comment]
Hi Yosida,
Thanks for the feedback!

- I have changed the createjob signature to return an error (instead of string,error). This is because jenkins only returns HTML when /createitem is invoked. (According to the docs the HTTP status code should be used to determine whether the operation is successful or not.)
- I have improved the createjob test case:
   - The test asserts that the job exists in the GetJobs response
   - The test generates a 'unique' job name to build when the test runs 
    - The response from createjob is no longer ignored

**Comment by yosida95** [comment]
Hi Dennis,

Thanks for improving.  I reviewed your patch again and it was nice!
The cause of failing CreateJob on my environment was that I disabled maven plugin.   I'm sorry...

I'll merge this patch when all changes are meld into one commit.  Could you meld commits by 'git rebase'?

**Comment by DennisDenuto** [comment]
Thanks Yosida,

No problem. I have squashed the commits into one as you asked.

**Comment by yosida95** [comment]
Merged.
I appreciate your contribution!

File: jenkins.go [hunk 1]
```diff
@@ -107,6 +151,15 @@ func (jenkins *Jenkins) GetBuild(job Job, number int) (build Build, err error) {
 	return
 }
 
+// Create a new job
+func (jenkins *Jenkins) CreateJob(mavenJobItem MavenJobItem, jobName string) error {
```
Comment (alexellis) [inline jenkins.go]: Do you have a version for jobs which are not Maven types. i.e. Freestyle?

**Pull request merged**